frontend/src/metabase/querying/notebook/components/NotebookDataPicker/NotebookDataPicker.tsx

```diff
@@ -203,29 +200,24 @@ function LegacyDataPicker({
   stageIndex,
   table,
   placeholder,
-  hasMetrics,
   isDisabled,
   onChange,
 }: LegacyDataPickerProps) {
-  const databaseId = Lib.databaseID(query);
   const tableInfo =
     table != null ? Lib.displayInfo(query, stageIndex, table) : undefined;
+  /**
+   * We try to blur the line between models and tables for embedding users.
+   * this property will change the way icons are displayed in the data picker trigger,
+   * so we need to remove it. Treating it as a table.
+   */
+  delete tableInfo?.isModel;
```

Comment:
![image](https://github.com/user-attachments/assets/426c76f9-0cc6-49b0-b5f9-d9545af5e81f)

See the [discussion on Slack](https://metaboat.slack.com/archives/C063Q3F1HPF/p1738267433382859?thread_ts=1738248655.057009&cid=C063Q3F1HPF) about this.

tl;dr we want to blur the line between models and tables so embedding users dno't have to worry about what they want to choose they can just treat everything as data.

Reply:
Deleting a property is not a way. Find another solution.